simulator/risc_v/t/unit_test.cpp

```diff
@@ -155,3 +155,45 @@ TEST_CASE( "RISCV lq/sq")
     memory->load_store( &load);
     CHECK( narrow_cast<uint64>( load.get_v_dst()) == 0xf);
 }
+
+template<typename T>
+struct TestData {
+    T src1, src2,dst;
+};
+
+TEST_CASE( "RISCV slo32")
+{
+    CHECK( RISCVInstr<uint32>( 0x20E797B3).get_disasm() == "slo $a5, $a5, $a4");
+    RISCVInstr<uint32> instr( "slo", 0);
+    std::vector<TestData<uint32>> cases = {
+        {0x1C, 2, 0x73},
+        {0xFFFFFFFF, 0xAA, 0xFFFFFFFF},
+        {0xAA, 0xFF, 0x7FFFFFFF},
+        {0xAB, 0xFF, 0xFFFFFFFF},
+    };
+    for (std::size_t i = 0; i < cases.size(); i++) {
+        instr.set_v_src( cases[i].src1, 0);
+        instr.set_v_src( cases[i].src2, 1);
+        instr.execute();
+        INFO( "Iteration: " << i);
+        CHECK( instr.get_v_dst() == cases[i].dst);
+    }
+}
+
+TEST_CASE ("RISCV slo64") {
+    RISCVInstr<uint64> instr( "slo", 0);
+    std::vector<TestData<uint64>> cases = {
+        {0x1C, 2, 0x73},
+        {0xFFFFFFFF, 5, 0x1FFFFFFFFF},
+        {0xFFFFFFFFFFFFFFFF, 0xFF, 0xFFFFFFFFFFFFFFFF},
+        {0xAA, 0xFF, 0x7FFFFFFFFFFFFFFF},
+        {0xAB, 0xFF, 0xFFFFFFFFFFFFFFFF},
```

Comment:
There are too many 'F"s.
I suggest to use our macro constants (all_ones<T> and msb_set<T>) for clarity.